pyanaconda/storage/checker.py

```diff
@@ -411,6 +412,51 @@ def verify_mounted_partitions(storage, constraints, report_error, report_warning
                                "installation. Please unmount it and retry.") % part.path)
 
 
+def verify_lvm_destruction(storage, constraints, report_error, report_warning):
+    """ Verify that destruction of LVM devices is correct.
+
+    This means: If a PV is destroyed, the VG on it must be destroyed too.
+    Additionally, destroying an incomplete VG warns about this.
```

Comment:
Maybe I'm misunderstanding something but this seems incorrect to me.

In general you should be able to destroy PV and just make VG smaller. I'm not sure if we support this in Anaconda, thought. The main problem we are trying to solve is that you have to have a complete VG (with all PVs) to be able to work with it. Am I missing something? Is this some kind of simplification?

Reply:
Yes, this is a simplification, at least compared to what you are thinking about. This does not check if there is no possible sequence of actions that could satisfy the high-level objectives. Instead, this checks if we can do what we do - that the selected disks can be just wiped. So the check is, will we leave a known dangling PV behind?

The comment was half-truth, so I changed it anyway. Now it says: A VG and all its PVs must be all destroyed together, nothing is allowed to remain.

As an aside, the action you are hinting at is not very useful, IMO. Effectively, you would have to successfully downsize the filesystems on all the LVs so as to fit on the remaining PVs. Is that possible? Yes. But IMO most users don't want a magical shrinking of their previous installation, but rather wipe everything and start anew.

Reply:
I changed the comment to match the final logic.